.github/workflows/deploy.yml

```diff
@@ -0,0 +1,41 @@
+name: Deploy
+
+on:
+  push:
+    tags:
+      - '*'
+  release:
+    types:
+      - published
+
+jobs:
+  deploy:
+    name: Publish package to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Wait for tests to succeed
+        uses: fountainhead/action-wait-for-check@v1.0.0
+        id: wait-for-ci
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          checkName: push-success
```

Comment:
(and success here too)

Reply:
Deploy only runs when publishing a new release. It must check the CI correctly passed on the HEAD of master. The CI on master run because of a `push` event when merge in PR, thus the associated CI has `push-success` name.

Reply:
EDIT: I changed it for `success`